researchseminarsdotorg.py

```diff
@@ -72,31 +72,67 @@ def add_talk_to_series(series_id, payload, authorization):
     else:
         return "", r.status_code
 
-
 def publish_to_researchseminars(talk):
     # talk should be provided in yaml format
     api_token = os.getenv("RESEARCHSEMINARS_API_TOKEN")
-    authorization = "vsf@virtualscienceforum.org %s" % api_token
+    authorization = "researchseminars@antonakhmerov.org %s" % api_token
+
+    # Set series ID
+    series_id = "none"
+    if( talk["event_type"] == "speakers_corner" ):
+        series_id = "speakerscorner"
+    if( talk["event_type"] == "lrc" ):
+        series_id = "VSFLRC"
+
+    # Check that we have a valid series_id
+    if( series_id == "none" ):
+        print("Invalid series_id for publishing to researchseminars")
+        return
 
-    # Find speakers' corner series, and create it if it doesn't exist
-    if not find_seminar_series("speakerscorner"):
-        print("[ResearchSeminars.org]: The speakerscorner seminar series "\
-              "does not yet exist. Creating it now")
-        create_seminar_series(SPEAKERS_CORNER_SEMINAR_SERIES)
+    # Find the series, and create it if it doesn't exist
+    if not find_seminar_series(series_id):
+        create_seminar_series(SPEAKERS_CORNER_SEMINAR_SERIES, authorization)
 
-    # TODO: Figure out if we need to edit the series;
-    #       Would be annoying since edits have to be approved
+    # TODO: If the series did not exist, it's creation has to be approved.
+    #       That means the rest of this code will fail on the first attempt.
 
     # Set up payload for talk creation
-    talk_payload = {"title":talk.get('title'),
-                    "speaker":talk.get('author'), # TODO: will be 'speakerS'
-                    "online": True,
-                    "start_time":talk["time"],
-                    "timezone":"UTC"
-                    }
+    talk_payload = {
+        "series_id": series_id,
+        "language":"en",
+        # Speaker info
+        "speaker":talk.get('speaker_name'),
+        "speaker_email":talk.get('email'),
+        "speaker_affiliation":talk.get('speaker_affiliation', 'Unknown'),
+
+        # Talk info
+        "title":talk.get('title'),
+        "abstract":talk.get('abstract'),
+        "start_time":talk["time"].strftime('%Y-%m-%dT%H:%M:%S'),
+        "end_time":(talk["time"] + timedelta(hours=1)).strftime('%Y-%m-%dT%H:%M:%S'),
+        "timezone":"UTC", # Not required per se, copied from seminar series
+
+        # Zoom info
+        "online": True,
+        "access_control":5, # Manual registration
+        "access_registration":talk.get('registration_url'),
+
+        # Extra info
+        # ! Leave these out if unavailable
+        #"slides_link":"http://Unavailable.org",
+        #"video_link":"http://ToBeUpdated",
+    }
+
+    # If a paper link is available, add it. Not all talks have preprints
+    if( talk.get('preprint', '') != '' ):
+        talk_payload["paper_link"] = "https://arxiv.org/abs/"+talk.get('preprint')
```

Comment:
We had an example of somebody using a DOI and a different repository, so this may produce nonsense.

I suggest using something like [this](https://github.com/virtualscienceforum/virtualscienceforum/blob/master/bots/render_talks.py#L21-L22).

Reply:
Thanks, I thought those would have been filtered before getting to this point. I was wrong. 
I assume you mean that we 

- use the arxiv link if talk['preprint'] is an arxiv identifier, 
- use talk['preprint'] as long as it is a valid URL,
- otherwise add this as a talk comment? 
 
The researchseminars API will fail if this entry is not a proper URL, and we can't guarantee that for talk['preprint'], right?

Reply:
We could regex-match a DOI, however I propose that we specifically regex match an arxiv preprint, and ignore anything else.

Reply:
Agreed 👍